src/rsa/padding.rs

```diff
@@ -269,42 +265,36 @@ impl RsaEncoding for PSS {
 
         // Step 3 is done by `PSSMetrics::new()` above.
 
-        // Step 4.
-        let mut salt = [0u8; MAX_SALT_LEN];
-        let salt = &mut salt[..metrics.s_len];
-        rng.fill(salt)?;
-
-        // Step 5 and 6.
-        let h_hash = pss_digest(self.digest_alg, m_hash, salt);
+        let (db, digest_terminator) = em.split_at_mut(metrics.db_len);
+        let h;
+        {
```

Comment:
Nit: `h` could be assigned to the output of this block to potentially clarify at what point its bound when scrolling the code.

Reply:
Good point. I actually just eliminated the scoping block as it's no longer important with the refactored code.

Reply:
That works too. I think its much easier to follow whats happening now 👍